File: aziz/python/ch04/bit-divide.py


def divide(x, y):
    result = 0

    power = 32

    yPower = y << power

    while x >= y:
        while yPower > x:
            yPower >>= 1
            power -= 1

        print "result %d, power %d, yPower %d" % (result, power, yPower)
        result += 1 << power
        x -= yPower


    return result



print divide(125, 5)

File: aziz/php/ch04/bit-divide.php
<?php

function divide($x, $y) {
    $result = 0;

    $power = 32;

    $yPower = $y << $power;

    while ($x >= $y) {
        while ($yPower > $x) {
            $yPower >>= 1;
            -- $power;
        }
        $result += 1 << $power;
        $x -= $yPower;
    }

    return $result;
}


var_dump(divide(9, 3));
var_dump(divide(99, 5));